### DISCUSSION TITLE
Apostrophe in star comment breaking the color code and readability

### DISCUSSION THREAD

**Discussion by amalcor** [krvajal/vscode-fortran-support#856] [Q&A - deprecated see Fortran Discourse]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

Hello,
I'm quite new to fotran, and currently facing two problems with the fortran extension a lot of comment with an apostrophe : ' character and also starting with a star : *.

If a comment like that exists in a F90 file, it is not recognized as a command, and the apostrophe breaks the readability of the rest of the file :
`
*-----   Excuse me, I'm a simple comment
`

Is this a bug or issue of the plugin, or are there any tricks to fix this issue locally, and activate the parsing of * starting line as comment in F90 files ?

Exemples : 

`      program hello
!---- Test'
        print *, 'Hello World'
      end program hello
`
works but : 

`
      program hello
*---- Test'
        print *, 'Hello World'
      end program hello
`
don't

### Expected Behaviour

Appropriate formatting

### Version of Modern Fortran

v3.4.2023032301

### Version of Visual Studio Code

v1.76

### Platform and Architecture

Linux
![image](https://user-images.githubusercontent.com/1580126/227192420-f9096dc0-ebfc-4df4-831d-46b3bb4424bd.png)

**Comment by gnikit** [comment]
That's because `*` are not standard comment characters, the `!` is

**Comment by amalcor** [comment]
Ok thanks for your reply. Would you advise it is not pertinent to implement anything regarding my issue in the module, and only fix this locally ? I suppose I would need to change fortran_free-form.tmLanguage.json lines 1255 to 1259 from :

    "comments": {
      "name": "comment.line.fortran",
      "begin": "!",
      "end": "(?=\\n)"
    },

to 

    "comments": {
      "name": "comment.line.fortran",
      "begin": "!|\*",
      "end": "(?=\\n)"
    },

**Comment by gnikit** [comment]
`*` is a reserved operator and format selector. I am relatively certain that the TextMate grammar regular expressions would struggle to distinguish between

```
a_cal = a * b ** 2 * a comment
```

The snippet that you provided btw is Fixed Form, not FreeForm. There are different syntax highlighting rules for Fixed Form, which work perfectly fine for your case

![image](https://user-images.githubusercontent.com/16143716/227204127-f7ec47ef-499b-4582-8429-bc8ebc2fcdd7.png)

You should set the `Language Mode` for your file to `Fortran77` (which is an alias for FortranFixedForm)

**Comment by amalcor** [comment]
Ha yes, I tried my change and it made line of multiplication screwed up.

I have a done a change of `Language mode` to Fortran77 for my F90 files, and it fixed a part of the problems. Now it seems that it's the fortls linter that is not using the Fixed Form, but there also some strange issue where an imported module is underlined in red, but ctrl+click leads to the appropriate file...  Some file also start with a line with a capital case C for a comment, it seems fortls analysis struggle with it.
